main.js

```diff
@@ -45,48 +45,101 @@ for (let i = 0; i < array3.length; i++) {
 }
 
 // Функция palindrome
-function isPalindrome (str) {
+function isPalindrome(str) {
     let strCompare = str.split('').reverse().join('');
     if (strCompare.toLowerCase === str.toLowerCase) {
         return true;
     } else {
         return false;
     }
 }
-console.log(isPalindrome ('Zorroz'));
+console.log(isPalindrome('Zorroz'));
 
 // Функция min(a, b) и функция max(a,b)
-function min (a, b) {
+function min(a, b) {
     if (a < b) {
         return a;
     } else {
         return b;
     }
 }
-console.log (min (1, 3));
+console.log(min(1, 3));
 
-function min (a, b){
-	return (a < b) ? a: b;
+function min(a, b) {
+    return (a < b) ? a : b;
 }
-console.log (min (-1, 3));
+console.log(min(-1, 3));
 
-function max (a, b) {
+function max(a, b) {
     if (a > b) {
         return a;
     } else {
         return b;
     }
 }
-console.log (max (1, 3));
+console.log(max(1, 3));
 
-function max (a, b) {
-	return (a < b) ? a: b;
+function max(a, b) {
+    return (a < b) ? a : b;
 }
-console.log (max (-1, 3));
+console.log(max(-1, 3));
 
 // Замена элементов массива 
 let array4 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 10];
 let str = array4.join(" ");
 let rep = str.replace(/0/gi, 'zero');
 let res = rep.split(" ");
 console.log(res);
+
+// Напишите функцию sum, которая возвращает сумму чисел следующим образом:
+function curry(f) {
+    return function (a) {
+        return function (b) {
+            return f(a, b);
+        };
+    };
+};
+function corSum(a, b) {
+    return a + b;
+};
+let sum = curry(corSum);
+console.log(sum(5)(2));
+module.exports = corSum;
+
+// Покрасьте абзацы по клику (событие click):
+const colors = ['magenta', 'cyan', 'firebrick', 'springgreen', 'skyblue'];
+const text1 = document.getElementById('text1');
+const text2 = document.getElementById('text2');
+const text3 = document.getElementById('text3');
+function firstColor(element) {
+    element.style.color = colors[0];
+};
+function changeColor(element) {
+    if (element.style.color === colors[0]) {
+        element.style.color = colors[1];
+    } else if (element.style.color === colors[1]) {
+        element.style.color = colors[2];
+    } else if (element.style.color === colors[2]) {
+        element.style.color = colors[3];
+    } else if (element.style.color === colors[3]) {
+        element.style.color = colors[4];
+    } else element.style.color = colors[0];
+};
```

Comment:
it's too bad resolve way - imagine that COLORS array has 1000 elements - is it clear to use 1000 of if's block?)

Reply:
use currentColorIndex = colors.indexOf(element.style.color) and return newColor => colors[currentColorIndex + 1]